sematic/config/settings.py

```diff
@@ -230,59 +230,82 @@ def as_bool(value: Optional[Any]) -> bool:
     return bool(distutils.util.strtobool(str_value))
 
 
-def _normalize_enum(enum_type: Type[AbstractSettingsVar], obj: Any):
-    """
-    Returns a value belonging to the specified enum which corresponds to the specified
-    object, if possible.
-    """
-    if isinstance(obj, enum_type):
-        return obj
-
-    try:
-        return enum_type[str(obj)]
-    except KeyError:
-        return None
-
-
-def _load_settings(file_path: str, vars: Type[AbstractSettingsVar]) -> Settings:
+def _load_settings(file_path: str) -> Settings:
```

Comment:
When you load settings for a plugin, can you:
- get what version of the plugin the settings were last written with
- get what version of the plugin has currently been loaded
- if the settings that were written are from a different major version than the current loaded plugin, surface this to the user

Surfacing to the user should probably be at least a warning, but could be an error. Later we could leverage this to allow plugins to automatically upgrade configs when users upgrade the plugin.

Reply:
Done.